app/assets/src/components/visualizations/heatmap/Heatmap.js

```diff
@@ -935,14 +1133,30 @@ export default class Heatmap {
     columnMetadataLabelEnter
       .append("rect")
       .attr("class", cs.hoverTarget)
-      .attr("width", this.rowLabelsWidth)
+      .attr("x", -this.options.marginLeft)
+      .attr("width", d => {
+        const xOffset =
```

Comment:
Maybe we can factor this logic out into a function, since it is being called in several different situations.